### PR TITLE
process_collector: collect received/transmitted bytes

### PR DESCRIPTION
**Pull request by huwcbjones**
### Describe your PR
On systems with procfs, collect the count of bytes received/transmitted of the process.

### What type of PR is this?
/kind enhancement

### Changelog Entry
```release-note
process collector: on systems with procfs report the count of bytes received/transmitted.
```

### CODE REVIEW

File: prometheus/process_collector.go [hunk 1]
```diff
@@ -100,6 +101,16 @@ func NewProcessCollector(opts ProcessCollectorOpts) Collector {
 			"Start time of the process since unix epoch in seconds.",
 			nil, nil,
 		),
+		inBytes: NewDesc(
+			ns+"process_net_bytes_in",
+			"Number of bytes received by the process.",
```
Comment (kakkoyun) [inline prometheus/process_collector.go:106]:
```suggestion
			"Number of bytes received by the process over the network.",
```

File: prometheus/process_collector.go [hunk 2]
```diff
@@ -100,6 +101,16 @@ func NewProcessCollector(opts ProcessCollectorOpts) Collector {
...
+			nil, nil,
+		),
+		outBytes: NewDesc(
+			ns+"process_net_bytes_out",
+			"Number of bytes sent by the process.",
```
Comment (kakkoyun) [inline prometheus/process_collector.go:111]:
```suggestion
			"Number of bytes sent by the process over the network.",
```

**Review by kakkoyun — APPROVED** [review]
LGTM! 🚀 
Thanks for the contribution.
I've just a nit on the description of the new metrics.

**Comment by huwcbjones** [comment]
Thanks for the feedback on the descriptions, I struggled conjuring up some succinct words!

**Comment by huwcbjones** [comment]
Ok, I thought the CI failure was a me issue, but no it appears it's failing in the same way on main.
See https://github.com/prometheus/client_golang/actions/runs/9907802012/job/27372245542 which ran on https://github.com/prometheus/client_golang/commit/b0809618ea20a3129e30fea63b01f16c623efc69

File: prometheus/process_collector.go [hunk 3]
```diff
@@ -100,6 +101,16 @@ func NewProcessCollector(opts ProcessCollectorOpts) Collector {
 			"Start time of the process since unix epoch in seconds.",
 			nil, nil,
 		),
+		inBytes: NewDesc(
+			ns+"process_net_bytes_in",
+			"Number of bytes received by the process over the network.",
+			nil, nil,
+		),
+		outBytes: NewDesc(
+			ns+"process_net_bytes_out",
```
Comment (SuperQ) [inline prometheus/process_collector.go:110]: This should follow Prometheus naming conventions.

```suggestion
			ns+"process_network_received_bytes_total",
			"Number of bytes received by the process over the network.",
			nil, nil,
		),
		outBytes: NewDesc(
			ns+"process_network_transmitted_bytes_total",
```

**Review by SuperQ — CHANGES_REQUESTED** [review]
Please follow Prometheus naming conventions.

File: prometheus/process_collector.go (hunk 3 above)
Comment (huwcbjones) [reply prometheus/process_collector.go]: Have tweaked naming, left it as in/out because that maps closer to where the actual value comes from.

Comment (SuperQ) [reply prometheus/process_collector.go]: in/out does not match our naming conventions.

Comment (huwcbjones) [reply prometheus/process_collector.go]: Then please link me the naming convention documentation.

Comment (SuperQ) [reply prometheus/process_collector.go]: https://prometheus.io/docs/practices/naming/

This one specifically is about ambiguity. In network terminology we talk about receive and transmit. For example, the node_exporter standard metrics are:

```
node_network_receive_bytes_total
node_network_transmit_bytes_total
```

We should follow that convention.

**Review by SuperQ — APPROVED** [review]

**Comment by huwcbjones** [comment]
> I guess we need to figure out why tests in main are broken. 😢

go 1.22 recently added the new runtime metric `go_godebug_non_default_behavior_netedns0_events_total`

**Comment by SuperQ** [comment]
Thanks for the tip, fixing in https://github.com/prometheus/client_golang/pull/1562.

**Comment by kakkoyun** [comment]
@huwcbjones Could please rebase? Let's make sure CI is green, and we can merge it.

**Review by kakkoyun — APPROVED** [review]

**Pull request merged**